bluebutton-data-pipeline-fhir-load/src/test/java/gov/hhs/cms/bluebutton/datapipeline/fhir/transform/DataTransformerTest.java

```diff
@@ -686,7 +814,91 @@ public void transformInsertPartDEventNotCompound() throws FHIRException {
 		ExplanationOfBenefit eob = (ExplanationOfBenefit) eobEntry.getResource();
 
 		ItemsComponent rxItem = eob.getItem().stream().filter(i -> i.getSequence() == 1).findAny().get();
-		Assert.assertEquals("RXDINV", rxItem.getType().getCode());
+		Assert.assertEquals("RXCINV", rxItem.getType().getCode());
+	}
+
+	/**
+	 * Verifies that the {@link DataTransformer} correctly uses the
+	 * {@link DataTransformer#CODED_ADJUDICATION_PART_D_NONCOVERED_SUPPLEMENT
+	 * code when the drug coverage status code =
+	 * {@link PartDEventRow#DRUG_CVRD_STUS_CD_SUPPLEMENT}
+	 * 
+	 * @throws FHIRException
+	 */
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	@Test
+	public void transformInsertPartDEventNonCoveredSupplement() throws FHIRException {
```

Comment:
I'm not sure if what I am going to include as an update is exactly what you had in mind, but it just involves putting into a helper method all the boilerplate code that goes into getting a Bundle. If it looks good, you should be able to use it in your tests as well.